src/firstpass.rs

```diff
@@ -1183,6 +1235,55 @@ impl<'a, 'b> FirstPass<'a, 'b> {
             None
         }
     }
+
+    /// Parses an attribute block content, such as `.class1 #id .class2`.
+    ///
+    /// Returns `(id, classes)`.
+    ///
+    /// It is callers' responsibility to find opening and closing characters of the attribute
+    /// block.
+    ///
+    /// If `last_ix` is less than `first_index`, attribute block is considered as empty.
+    ///
+    /// Note that this parsing requires explicit whitespace separators between
+    /// attributes. This is intentional design with the reasons below:
+    ///
+    /// * to keep conversion simple and easy to understand for any possible input,
+    /// * to avoid adding less obvious conversion rule that can reduce compatibility
+    ///   with other implementations more, and
+    /// * to follow the major design of implementations with the support for the
+    ///   attribute blocks extension (as of this writing).
+    ///
+    /// See also: [`Options::ENABLE_HEADING_ATTRIBUTES`].
+    ///
+    /// [`Options::ENABLE_HEADING_ATTRIBUTES`]: `crate::Options::ENABLE_HEADING_ATTRIBUTES`
+    fn parse_inside_attribute_block(
+        &self,
+        first_ix: usize,
+        last_ix: usize,
+        id: &mut Option<CowStr<'a>>,
+        classes: &mut Vec<CowStr<'a>>,
+    ) {
+        // Return earlily when the attribute block is empty.
```

Comment:
Nit: small typo

Reply:
Thank you!
I've never realized that the word "[earlily](https://en.wiktionary.org/wiki/earlily)" is considered as _uncommon_...
Learning foreign languages is hard! 😭